srv/salt/_modules/rgw.py

```diff
@@ -230,6 +230,11 @@ def endpoints(cluster='ceph'):
 
 
 def s3connect(user):
+    """
+    Return an S3 connection
+    """
+    if access_key(user) is None or secret_key(user) is None:
+        return
     endpoint = endpoints()[0]
 
     s3conn = boto.connect_s3(
```

Comment:
@theanalyst That's the boto(3) call we have to adapt, right?

Reply:
@jschmid1connect_s3 complaints only when None is passed for access_key/secret_key(https://github.com/SUSE/DeepSea/issues/910). So exit early in that case.

Reply:
yes, for newer versions we can move this to boto3